thamos/cli.py

```diff
@@ -1379,4 +1366,93 @@ def support(output: str) -> None:
             json.dump(info, output_file, sort_keys=True, indent=2)
 
 
+@cli.command("whatprovides")
+@click.pass_context
+@click.option(
+    "--output-format",
+    "-o",
+    type=click.Choice(["json", "yaml", "table"]),
+    default="table",
+    help="Specify output format for the status report.",
+)
+@click.argument("import_name", type=str, required=True)
+@handle_cli_exception
+def whatprovides(import_name: str, output_format: str) -> None:
+    """For a given import_name returns list of (package_name, package_version, index_url) triplets.
+
+    Examples:
+      thamos whatprovides sklearn
+
+    """
+    _LOGGER.info("Returning information on package %r", import_name)
+    result = get_package_from_imported_packages(import_name)
+
+    if output_format == "yaml":
+        yaml.safe_dump({"package_names": result}, sys.stdout)
+    elif output_format == "json":
+        json.dump({"package_names": result}, sys.stdout, indent=2)
+        sys.stdout.write("\n")
+    elif output_format == "table":
+        table = Table()
```

Comment:
It should be possible to call directly `_print_report` here. It will save this code and reuse the one used when printing other parts to users. It should also adjust table names, so underscores are not listed.

Reply:
![Screenshot from 2021-11-23 09-38-31](https://user-images.githubusercontent.com/27498679/142992670-a4b3a655-098b-4a7d-ac0b-19e0d3eb7916.png)

updated! Thanks!